proposals/0000-swiftpm-script-support.md

```diff
@@ -0,0 +1,156 @@
+
+The Swift community has also recognized and tried to provide better scripting support through projects like [Marathon](http://github.com/johnsundell/marathon) and [swift-sh](http://github.com/mxcl/swift-sh). These projects leverage the Swift Package Manager to fetch and build external packages referenced in a Swift script. We believe we can take this approach further and provide a much better scripting experience by improving the integration between the Swift language and the Swift Package Manager. This will also allow Swift users to easily share their scripts without requiring to install additional tooling.
+
+This proposal would dramatically improve the developer experience for writing scripts in Swift. Dependencies for common use-cases would be easy to integrate and scripts would be portable across machines without additional tools installed on the host machine. 
+
+## Proposed solution
+
+We propose to add a new attribute `@package` to import statements for referencing Swift packages. This attribute would be only valid when running in the Swift interpreter (via `swift MyScript.swift` or `swift run MyScript.swift`) and will be rejected if used during compilation of a regular Swift module. Swift Package Manager will then fetch and build the declared packages and make them available to the script when executing it.
+
+```swift
+@package(url: "https://github.com/jpsim/Yams.git", from: "2.0.0")
+import Yams
+```
+
+This script can be executable by either `swift MyScript.swift` or `swift run MyScript.swift` which will utilize SwiftPM's dependency fetching and resolution logic.
+
+The `@package(...)`syntax will allow the same parameters as SwiftPM's [Package Dependency Description](https://docs.swift.org/package-manager/PackageDescription/PackageDescription.html#package-dependency) which should feel natural to developers familiar with Swift Package Manager manifest syntax.  
+
+## Detailed design
+
+
+#### Basic usage
+The most common case is expected to be importing a dependency where the product name is equivalent to the imported module name.
+
+Specifying the version of the package is optional. If omitted, SwiftPM will resolve the latest version of the dependency.
+```swift
+@package(url: "https://github.com/jpsim/Yams.git")
+import Yams
+```
+
+When a dependency product and target names do not match, the product must be explicitly specified.
+
+For an example, lets look at the Package description below for `swift-tools-support-core`. 
+```swift
+let package = Package(
+    name: "swift-tools-support-core",
+    products: [
+        .library(
+            name: "SwiftToolsSupport",
+            type: .dynamic,
+            targets: ["TSCBasic", "TSCUtility"]),
+    ],
+    // Rest of Package.swift truncated for brevity .....
+}
+```
+
+With the above package description, the `SwiftToolsSupport` product does not match a target within the package description so it must be specified explicitly.
```

Comment:
The following feels more correct: "With the above package description, the exported module names and product names don't match, so the product must be specific explicitly."